core/src/main/scala/org/bitcoins/core/crypto/TxSigComponent.scala

```diff
@@ -166,6 +180,17 @@ object BaseTxSigComponent {
 
 }
 
+object P2SHTxSigComponent {
+
+  def apply(
+      transaction: Transaction,
```

Comment:
This should probably be typed as a `: BaseTransaction`

Reply:
This will require us to have to do something like `P2SHTxSigComponent(unsignedTx.asInstanceOf[BaseTransaction],...` for every call in `Signer.scala`

Reply:
Is it not possible for this to be a `WitnessTransaction` though? (If there are other outputs or inputs that use segwit)

Reply:
You are right :+1: